astroplan/scheduling.py

```diff
@@ -124,8 +136,159 @@ def components(self, val):
 
         self._components = val
         self.duration = duration
+        
+    @classmethod
+    @u.quantity_input(duration=u.second)
+    def from_duration(cls, duration):
+        # for testing how to put transitions between observations during
+        # scheduling without considering the complexities of duration
+        tb = TransitionBlock({None: 0*u.second})
+        tb.duration = duration
+        return tb
+
+
+class Schedule(object):
+    """
+    An object that represents a schedule, consisting ofa list of
+    `~astroplan.scheduling.Slot` objects
+    """
+    # as currently written, there should be no consecutive unoccupied slots
+    # this should change to allow for more flexibility (e.g. dark slots, grey slots)
+    
+    def __init__(self, start_time, end_time, constraints=None):
+        """
+        Parameters:
+        -----------
+        start_time : `~astropy.time.Time`
+            The starting time of the schedule; the start of your
+            observing window
+        end_time : `~astropy.time.Time`
+           The ending time of the schedule; the end of your
+           observing window
+        constraints : sequence of `Constraint`s
+           these are constraints that apply to the entire schedule
+        """
+        self.slots = [Slot(start_time, end_time)]
+        self.constraints = constraints
+        self.slew_duration = 4*u.min
+        # TODO: replace/overwrite slew_duration with Transitioner calls
+
+    def __repr__(self):
+        return 'Schedule containing ' + str(len(self.observing_blocks)) + \
+               ' observing blocks between ' + str(self.slots[0].start.iso) + \
+               ' and ' + str(self.slots[-1].end.iso)
+
+    def apply_constraints(self):
```

Comment:
I think that would be good, if possible. From a conceptual point of view, I think changing a ```Schedule``` should be the job of the ```Scheduler``` and not the ```Schedule``` itself. Otherwise the two overlap in functionality in a potentially confusing way IMO.